monai/apps/manifest/reference_resolver.py

```diff
@@ -105,42 +111,46 @@ def _resolve_one_item(self, id: str, waiting_list: Optional[Set[str]] = None):
         waiting_list.add(id)
 
         ref_ids = self.find_refs_in_config(config=item_config, id=id)
-
-        # if current item has reference already in the waiting list, that's circular references
         for d in ref_ids:
+            # if current item has reference already in the waiting list, that's circular references
             if d in waiting_list:
                 raise ValueError(f"detected circular references for id='{d}' in the config content.")
-
-        # # check whether the component has any unresolved references
-        for d in ref_ids:
+            # check whether the component has any unresolved references
             if d not in self.resolved_content:
                 # this referring item is not resolved
-                if d not in self.items:
-                    raise ValueError(f"the referring item `{d}` is not defined in the config content.")
+                try:
+                    look_up_option(d, self.items, print_all_options=False)
+                except ValueError as err:
+                    raise ValueError(f"the referring item `@{d}` is not defined in the config content.") from err
                 # recursively resolve the reference first
-                self._resolve_one_item(id=d, waiting_list=waiting_list)
+                self._resolve_one_item(id=d, waiting_list=waiting_list, **kwargs)
+                waiting_list.remove(d)
```

Comment:
added this `waiting_list.remove(d)` otherwise there is a bug when `_resolve_one_item(id="")`